fadecandy_driver/src/ros/fadecandy_driver_ros.cpp

```diff
@@ -0,0 +1,83 @@
+#include "./fadecandy_driver_ros.h"
+#include <fadecandy_msgs/LEDArray.h>
+
+namespace fadecandy_driver
+{
+FadecandyDriverRos::FadecandyDriverRos()
+{
+  ros::NodeHandle nh;
+
+  timer_ = nh.createTimer(ros::Duration(1.), &FadecandyDriverRos::timerCallback, this);
+
+  diagnostic_updater_.add("Info", this, &FadecandyDriverRos::diagnosticsCallback);
+
+  led_subscriber = nh.subscribe<fadecandy_msgs::LEDArray>("/set_leds", 1, &FadecandyDriverRos::setLedsCallback, this);
+  initialized = false;
+}
+
+void FadecandyDriverRos::connect()
+{
+  if (fadecandy_device != NULL)
+  {
+    release();
+    fadecandy_device = NULL;
+  }
+
+  if (!initialized)
+  {
+    initialized = FadecandyDriverRos::intialize();
+    if (initialized)
+    {
+      ROS_INFO("Connected to Fadecandy device");
+      unsigned char serial_number[64];
```

Comment:
please use std::strings for this

Reply:
This cannot be because this variable is used in the function  below `libusb_get_string_descriptor_ascii` which takes only unsigned char as input variable

Reply:
`.data()` but this should be handled in the driver, not the wrapper.

Reply:
@reinzor yes! it's already moved to the driver with the latest commit